src/keys/secret_key.rs

```diff
@@ -26,6 +27,24 @@ pub enum SecretKey {
     BlsShare(SerdeSecret<threshold_crypto::SecretKeyShare>),
 }
 
+impl SecretKey {
+    /// Construct a secret key from a hex string
+    ///
+    /// Similar to public key, it is often useful in user
+    /// facing apps to be able to set your own secret
+    /// key without depending on both the ed25519_dalek
+    /// and hex crates just to reimplement this function
+    pub fn ed25519_from_hex(hex: &str) -> Result<Self> {
+        let bytes = hex::decode(hex).map_err(|_| {
+            Error::FailedToParse("Couldn't parse bytes from secret key hex".to_string())
```

Comment:
I think we can include the error `hex::decode` error in the message, and do this consistently in the other cases, including for the ones for public key. I like this one which has more context info since it's saying what has failed exactly when trying to decode it as a string, along with the hex decoding error.